### PR TITLE
Add info alert for In-transit Encryption in external mode

### PR DESCRIPTION
**Pull request by malayparida2000**
https://issues.redhat.com/browse/RHSTOR-6581

The configurations for In-Transit Encryption in external mode are all required to be done on the RHCS cluster by the ceph admin. Ticking the checkbox on ODF side only sets the kernel mount option for cephFS. So it might be misleading the user to think, the checkbox is enough to enable Eit in external mode. So added an info alert under the checkbox to clarify that settings need to be applied on the RHCS cluster. Also reworded the description of the checkbox to make it more clear to the user.

### CODE REVIEW

File: packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx [hunk 1]
```diff
@@ -282,7 +282,7 @@ export const Encryption: React.FC<EncryptionProps> = ({
             }
           />
         )}
-        {!isMCG && (
+        {!isMCG && !isExternal && (
```
Comment (parth-gr) [inline packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx:285]: Wasn't it already hiding?

And remove `!isExternal` will actually show it?

Comment (malayparida2000) [reply packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx:285]: 0cWZGEK
Can confirm the checkbox is visible today, from a old screenshot.

Comment (malayparida2000) [reply packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx:285]: I do not have a external mode cluster handy, The dev cluster we get are aws based & don't allow to reach the external mode setup page. Or else I could have tested the change too.

Comment (parth-gr) [reply packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx:285]: @SanjalKatiyar PTAL

**Comment by SanjalKatiyar** [comment]
We need few more changes now if we are removing in-transit from external mode:
1. Step is currently called `Security and network`, we need to talk to Eran/Chandan and rename this to maybe just `Network` now.
2. Also need to remove details from here: https://github.com/red-hat-storage/odf-console/blob/master/packages/odf/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx#L231-L239.

**Comment by parth-gr** [comment]
@malayparida2000 does this will require, external mode setup changes?

> `Ticking the checkbox on ODF side only sets the kernel mount option for cephFS.`

As currently we dont have a flag for  osd encryption.

**Comment by parth-gr** [comment]
@malayparida2000 i think we dont need to for internal mode also, as we made this by-default in both the modes downstream

**Comment by malayparida2000** [comment]
> @malayparida2000 i think we dont need to for internal mode also, as we made this by-default in both the modes downstream

No, Actually we made RequireMSGR2 default, Not Encryption. Encryption in transit is still an option to be choosen.

**Comment by SanjalKatiyar** [comment]
`instead the instructions should be covered by docs` >> @malayparida2000 can you please share the steps ?? this change could/might have an impact on another epic: https://issues.redhat.com/browse/RHSTOR-5286

**Comment by malayparida2000** [comment]
> `instead the instructions should be covered by docs` >> @malayparida2000 can you please share the steps ?? this change could/might have an impact on another epic: https://issues.redhat.com/browse/RHSTOR-5286

For enabling Encryption for External Mode
On RHCS Cluster-
```
# ceph config set global ms_client_mode secure
# ceph config set global ms_cluster_mode secure
# ceph config set global ms_service_mode secure
# ceph config set global rbd_default_map_options ms_mode=secure

Restart all ceph daemons (mgr, mon, osd, mds etc)
```

On ODF Cluster-
storagecluster CR
```
network:
  connections:
      encryption:
        enabled: true
        
 oc patch storagecluster ocs-storagecluster -n openshift-storage --type json --patch  '[{ "op": "replace", "path": "/spec/network", "value": {"connections": {"encryption": {"enabled": true}}} }]'
```

**Comment by malayparida2000** [comment]
> We need few more changes now if we are removing in-transit from external mode:
> 
> 1. Step is currently called `Security and network`, we need to talk to Eran/Chandan and rename this to maybe just `Network` now (if needed). Please make sure they are in loop.
> 2. Also need to remove details from here: https://github.com/red-hat-storage/odf-console/blob/master/packages/odf/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx#L231-L239.

Along with [1] removed another change I found.

**Comment by malayparida2000** [comment]
As I see we have 2 options,
1-Removing the UI checkbox from External Mode deployment flow
2-Adding a tooltip or note near the checkbox which clarifies that other settings need to be applied on the RHCS cluster to achieve Encryption In Transit

**Comment by SanjalKatiyar** [comment]
> > `instead the instructions should be covered by docs` >> @malayparida2000 can you please share the steps ?? this change could/might have an impact on another epic: https://issues.redhat.com/browse/RHSTOR-5286
> 
> For enabling Encryption for External Mode On RHCS Cluster-
> 
> ```
> # ceph config set global ms_client_mode secure
> # ceph config set global ms_cluster_mode secure
> # ceph config set global ms_service_mode secure
> # ceph config set global rbd_default_map_options ms_mode=secure
> 
> Restart all ceph daemons (mgr, mon, osd, mds etc)
> ```
> 
> On ODF Cluster- storagecluster CR
> 
> ```
> network:
>   connections:
>       encryption:
>         enabled: true
>         
>  oc patch storagecluster ocs-storagecluster -n openshift-storage --type json --patch  '[{ "op": "replace", "path": "/spec/network", "value": {"connections": {"encryption": {"enabled": true}}} }]'
> ```

steps seems  fairly straightforward to me, instead of removing checkbox from the external mode UI we can also consider just adding link to pre/post-requisite CLI steps (doc) and adding an info message on the UI. @malayparida2000 can we plz discuss this with Eran/Chandan ?? Or, is it already discussed and decided to move forward with removal of the checkbox ??

**Comment by SanjalKatiyar** [comment]
> `instead the instructions should be covered by docs` >> @malayparida2000 can you please share the steps ?? this change could/might have an impact on another epic: https://issues.redhat.com/browse/RHSTOR-5286

based on the steps shared here: https://github.com/red-hat-storage/odf-console/pull/1647#issuecomment-2426095403, looks like RHSTOR-5286 is **not** affected.

**Comment by malayparida2000** [comment]
> > > `instead the instructions should be covered by docs` >> @malayparida2000 can you please share the steps ?? this change could/might have an impact on another epic: https://issues.redhat.com/browse/RHSTOR-5286
> > 
> > 
> > For enabling Encryption for External Mode On RHCS Cluster-
> > ```
> > # ceph config set global ms_client_mode secure
> > # ceph config set global ms_cluster_mode secure
> > # ceph config set global ms_service_mode secure
> > # ceph config set global rbd_default_map_options ms_mode=secure
> > 
> > Restart all ceph daemons (mgr, mon, osd, mds etc)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > On ODF Cluster- storagecluster CR
> > ```
> > network:
> >   connections:
> >       encryption:
> >         enabled: true
> >         
> >  oc patch storagecluster ocs-storagecluster -n openshift-storage --type json --patch  '[{ "op": "replace", "path": "/spec/network", "value": {"connections": {"encryption": {"enabled": true}}} }]'
> > ```
> 
> steps seems fairly straightforward to me, instead of removing checkbox from the external mode UI we can also consider just adding link to pre/post-requisite CLI steps (doc) and adding an info message on the UI. @malayparida2000 can we plz discuss this with Eran/Chandan ?? Or, is it already discussed and decided to move forward with removal of the checkbox ??

There has been no discussion, This PR is just a proposal from my side to drive that discussion forward, whichever option from the2 I mentioned we find more preferable we can proceed with that.

**Comment by malayparida2000** [comment]
> There has been no discussion, This PR is just a proposal from my side to drive that discussion forward, whichever option from the 2 I mentioned we find more preferable we can proceed with that.

Created a slack thread for finalising the choice

https://ibm-systems-storage.slack.com/archives/C06SS96EECT/p1729503293391979

**Comment by parth-gr** [comment]
> Encryption in transit is still an option to be choosen

then we need to introduce a new flag in the python script, before skipping it here.

**Comment by malayparida2000** [comment]
> > Encryption in transit is still an option to be choosen
> 
> then we need to introduce a new flag in the python script, before skipping it here.

Flag for?

**Comment by parth-gr** [comment]
Instead of having this, user need to pass the details using a flag, and it should update the encryption.
` oc patch storagecluster ocs-storagecluster -n openshift-storage --type json --patch  '[{ "op": "replace", "path": "/spec/network", "value": {"connections": {"encryption": {"enabled": true}}} }]'`

**Comment by malayparida2000** [comment]
Acc to the discussion with Eran we would keep the checkbox with additional details about the required configurations in RHCS cluster.
Updated the PR accordingly.

**Comment by malayparida2000** [comment]
/retitle Change description of In-transit Encryption checkbox

**Comment by malayparida2000** [comment]
Internal Mode
Screenshot 2024-10-23 at 4 04 01 PM
External Mode
Screenshot 2024-10-23 at 4 16 28 PM

**Comment by SanjalKatiyar** [comment]
@malayparida2000 need to run `yarn i18n` command before commit else CI will fail.

**Comment by malayparida2000** [comment]
> @malayparida2000 need to run `yarn i18n` command before commit else CI will fail.

Thanks, Updated now.

**Comment by malayparida2000** [comment]
/retitle Add info alert for In-transit Encryption in external mode

**Comment by parth-gr** [comment]
@malayparida2000 so we are keeping the checkbox to update the storagecluster?

**Comment by malayparida2000** [comment]
> @malayparida2000 so we are keeping the checkbox to update the storagecluster?

Yes Parth according to the discussion we had with Eran & Chandan we would retain the checkbox with an added info alert.
Create-storagesystem (1)

**Comment by malayparida2000** [comment]
Screenshots from the new changes
Internal Mode
Screenshot 2024-11-05 at 3 18 34 PM
External Mode
Screenshot 2024-11-05 at 3 08 29 PM

File: packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx [hunk 2]
```diff
@@ -291,13 +297,31 @@ export const Encryption: React.FC<EncryptionProps> = ({
...
+          <Alert
+            variant={AlertVariant.info}
+            isInline
+            title={t(
+              'Verify your RHCS cluster has the necessary in-transit encryption settings configured to properly enable in-transit encryption on your external cluster.'
+            )}
+          >
+            {t('Refer to the documentation for detailed configuration steps.')}{' '}
+            <a
+              href="YOUR_DOCUMENTATION_LINK_URL" // Replace with actual doc link when available
```
Comment (parth-gr) [inline packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx:317]: can we link to ceph doc

File: packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx [hunk 3]
```diff
@@ -291,13 +297,31 @@ export const Encryption: React.FC<EncryptionProps> = ({
...
             )}
             onChange={(_event, checked: boolean) =>
               handleInTransitEncryptionOnChange(checked)
             }
           />
         )}
+        {isExternal && encryption.inTransit && (
+          <Alert
+            variant={AlertVariant.info}
+            isInline
```
Comment (SanjalKatiyar) [inline packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx:310]:
```suggestion
            isInline
            className='pf-v5-u-mt-sm'
```

File: packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx (hunk 2 above)
Comment (SanjalKatiyar) [inline packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx:317]: @malayparida2000 plz talk to doc team and get the skeleton link which we can add here something like for example: https://github.com/red-hat-storage/odf-console/blob/master/packages/ocs/dashboards/persistent-internal/status-card/status-card.tsx#L53-L62

File: packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx [hunk 4]
```diff
@@ -291,13 +300,35 @@ export const Encryption: React.FC<EncryptionProps> = ({
...
+            className="pf-v5-u-mt-sm"
+            title={
+              <>
+                {t(
+                  'Verify your RHCS cluster has the necessary in-transit encryption settings configured to enable in-transit encryption on your external cluster. Refer to the documentation for detailed configuration steps.'
+                )}{' '}
+                <a
+                  href={eitSettingsforRHCS(odfDocVersion)}
```
Comment (SanjalKatiyar) [inline packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx:321]:
```suggestion
                  href={editSettingsforRHCS(odfDocVersion)}
```

File: packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx (hunk 2 above)
Comment (malayparida2000) [reply packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx]: Added ref to skeleton doc link as suggested by Sanjal

File: packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx (hunk 4 above)
Comment (malayparida2000) [reply packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx]: gave a better name

File: packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx (hunk 2 above)
Comment (malayparida2000) [reply packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/encryption.tsx]: Added skeleton link, Thanks for the suggestion.

**Comment by malayparida2000** [comment]
Internal Mode & Provider Mode
Screenshot 2024-11-14 at 2 28 07 PM
External Mode
Screenshot 2024-11-14 at 2 22 58 PM

**Comment by malayparida2000** [comment]
> Internal Mode & Provider Mode Screenshot 2024-11-14 at 2 28 07 PM External Mode Screenshot 2024-11-14 at 2 30 16 PM

@Chandan141996 PTAL

File: packages/shared/src/constants/doc.ts [hunk 5]
```diff
@@ -14,3 +14,13 @@ export const odfDRDocApplyPolicy = (odfDocVersion) =>
   `${odfDRDocHome(
     odfDocVersion
   )}#apply-drpolicy-to-sample-application_manage-dr`;
+
+export const odfDeployExternalMode = (odfDocVersion) =>
+  `${odfDocBasePath(
+    odfDocVersion
+  )}/deploying_openshift_data_foundation_on_external_storage`;
```
Comment (SanjalKatiyar) [inline packages/shared/src/constants/doc.ts:21]: this is un-used, right ??

Comment (SanjalKatiyar) [reply packages/shared/src/constants/doc.ts:21]: can u remove it...

Comment (malayparida2000) [reply packages/shared/src/constants/doc.ts]: Updated

Comment (malayparida2000) [reply packages/shared/src/constants/doc.ts]: Just for reference we want this to point at 

`https://docs.redhat.com/en/documentation/red_hat_openshift_data_foundation/4.18/html/deploying_openshift_data_foundation_in_external_mode/deploy-openshift-data-foundation-using-red-hat-ceph-storage#creating-an-openshift-data-foundation-cluster-service-for-external-storage_ceph-external`

**Pull request merged**